### PR TITLE
🤸 Shorten FOUC, by using `link` instead of `@import` and `preload`

### PR DESCRIPTION
**Pull request by tomalec**
https://github.com/Starcounter/RebelsLounge/issues/244

### CODE REVIEW

**Comment by warpech** [comment]
TL;DR replace `<style>@import url "..."</style>` with both:

- `<link rel="preload" href"...">` before the first `<template>`
- `<link rel="stylesheet" href"...">` in shadow DOM

**Comment by tomalec** [comment]
Merge conflicts resolved

File: src/BlendingEditor/wwwroot/BlendingEditor/views/AreasPage.html [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ﻿<link rel="import" href="/sys/juicy-select/juicy-select.html">
+<link rel="preload" type="stylesheet" href="/sys/uniform.css/uniform.css">
```
Comment (warpech) [inline src/BlendingEditor/wwwroot/BlendingEditor/views/AreasPage.html]: Should be `link rel="preload" as="style"`, according to https://www.smashingmagazine.com/2016/02/preload-what-is-it-good-for/

Comment (tomalec) [reply src/BlendingEditor/wwwroot/BlendingEditor/views/AreasPage.html]: Fixed

**Comment by warpech** [comment]
One question though: is it possible to keep `link rel="preload"` in Shadow DOM?

**Comment by tomalec** [comment]
> is it possible to keep `link rel="preload"` in Shadow DOM?

What would it give? It will start downloading exactly same moment when regular `link`

**Comment by warpech** [comment]
> What would it give? It will start downloading exactly same moment when regular link

1. It would still dedupe the network request (`link rel="stylesheet"` does not dedupe the requests)
2. As the solution designer, you would be able to remove preloading of Uniform.css when you decide to remove Uniform classes in a custom composition

**Comment by tomalec** [comment]
> 1. It would still dedupe the network request (link rel="stylesheet" does not dedupe the requests)

But you will still need `<link rel="stylesheet">`. Please provide the code of what you mean in https://github.com/Starcounter/Blending/pull/102#issuecomment-369274025

**Comment by warpech** [comment]
Of course! As I wrote:

> is it possible to keep `link rel="preload"` in Shadow DOM? It makes sense to preload only when we know that the stylesheet is actually used.

I mean to do this:

```html
<template>
    foo
    <template is="declarative-shadow-dom">
        <link rel="preload" as="style" href="/sys/uniform.css/uniform.css">
        <link rel="preload" as="style" href="/BlendingEditor/css/blendinguniform.css">
        <link rel="stylesheet" href="/sys/uniform.css/uniform.css">
        <link rel="stylesheet" href="/BlendingEditor/css/blendinguniform.css">
        bar
    </template>
</template>
```

Instead of this:

```html
<link rel="preload" as="style" href="/sys/uniform.css/uniform.css">
<link rel="preload" as="style" href="/BlendingEditor/css/blendinguniform.css">
﻿<template>
    foo
    <template is="declarative-shadow-dom">
        <link rel="stylesheet" href="/sys/uniform.css/uniform.css">
        <link rel="stylesheet" href="/BlendingEditor/css/blendinguniform.css">
        bar
    </template>
</template>
```

**Comment by tomalec** [comment]
Then what is the benefit of keeping `<link rel="preload" as="style" href="/sys/uniform.css/uniform.css">` just before `<link rel="stylesheet" href="/sys/uniform.css/uniform.css">`?  AFAIK it will behave just like having `<link rel="stylesheet" href="/sys/uniform.css/uniform.css">` as we had before.

**Comment by warpech** [comment]
I wrote two benefits https://github.com/Starcounter/Blending/pull/102#issuecomment-369586119, are they wrong?

**Comment by tomalec** [comment]
1. Will not preload a thing, as you preload it exactly at the time you are loading it. and I don't think it will dedupe it either.

**Comment by tomalec** [comment]
However, it may be valid point that preloading it before the template is stamped, is not needed at all.

**Pull request merged**